sklearn/pipeline.py

```diff
@@ -253,6 +253,13 @@ def _fit(self, X, y=None, **fit_params):
         fit_params_steps = {name: {} for name, step in self.steps
                             if step is not None}
         for pname, pval in fit_params.items():
+            if '__' not in pname:
+                raise TypeError(
+                    "Passing a parameter directly to a pipeline isn't "
```

Comment:
I'm not sure if this is going to match the user's understanding.. Might be better to specify "Pipeline.fit does not accept the {} parameter. You can pass ..."